components/notebook-controller/Makefile

```diff
@@ -21,77 +13,125 @@ else
+
+##@ Build
 
-# Build the docker image
-docker-build: test
+.PHONY: manager
+manager: generate fmt vet ## Build manager binary.
+	go build -o bin/manager main.go
+
+.PHONY: run
+run: manifests generate fmt vet ## Run a controller from your host.
+	go run ./main.go
+
+.PHONY: docker-build
+docker-build: test ## Build docker image with the manager.
 	cd .. && docker build . -t ${IMG}:${TAG} -f ./notebook-controller/Dockerfile
-	@echo "updating kustomize image patch file for manager resource"
-	sed -i'' -e 's@image: .*@image: '"${IMG}:${TAG}"'@' ./config/default/manager_image_patch.yaml
 
-# Push the docker image
-docker-push:
+.PHONY: docker-push
+docker-push: ## Push docker image with the manager.
 	docker push ${IMG}:${TAG}
 
-# find or download controller-gen
-# download controller-gen if necessary
-controller-gen:
-ifeq (, $(shell which controller-gen))
-	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.0
-CONTROLLER_GEN=$(GOBIN)/controller-gen
-else
-CONTROLLER_GEN=$(shell which controller-gen)
-endif
+##@ Deployment
 
-# TODO(jlewi): Can we get rid of this and just use skaffold?
-build-gcr: test
-	docker build -t $(IMG):$(TAG) .
-	@echo Built $(IMG):$(TAG)
+ifndef ignore-not-found
+  ignore-not-found = false
+endif
 
-push-gcr: build-gcr
-	docker push $(IMG):$(TAG)
-	@echo Pushed $(IMG):$(TAG)
+.PHONY: install
+install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
+	kubectl get crd notebooks.kubeflow.org || $(KUSTOMIZE) build config/crd | kubectl create -f -
+
+.PHONY: uninstall
+uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
+	$(KUSTOMIZE) build config/crd | kubectl delete --ignore-not-found=$(ignore-not-found) -f -
+
+.PHONY: deploy
+deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	sed -i'' -e 's@newName: .*@newName: '"${IMG}"'@' ./config/base/kustomization.yaml
+	sed -i'' -e 's@newTag: .*@newTag: '"${TAG}"'@' ./config/base/kustomization.yaml
+	$(KUSTOMIZE) build config/base | kubectl apply -f -
+
+.PHONY: undeploy
+undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
+	$(KUSTOMIZE) build config/base | kubectl delete --ignore-not-found=$(ignore-not-found) -f -
+
+CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
+.PHONY: controller-gen
+controller-gen: ## Download controller-gen locally if necessary.
+	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.8.0)
+
+KUSTOMIZE = $(shell pwd)/bin/kustomize
+.PHONY: kustomize
+kustomize: ## Download kustomize locally if necessary.
+	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
```

Comment:
It's failing because there is a patch trying to modify the old v1beta1 CRD but there is no match (obviously):

```
patchesJson6902:
# Remove once the following issue is resolved:
# https://github.com/kubeflow/kubeflow/issues/5722
- path: patches/old_crd.yaml
  target:
    group: apiextensions.k8s.io
    version: v1beta1 # <--------- This should be v1
    kind: CustomResourceDefinition
    name: notebooks.kubeflow.org
 ```
 
Since https://github.com/kubeflow/kubeflow/issues/5722 is resolved I think the best solution is to remove the entire patch block, what do you think?

If we are using Kustomize 3.2 in manifests I can update the Makefile to download that version to be aligned, that may save some compatibility issues in the future.

Reply:
I agree to remove the patch altogether. The patch was making the validation more lax, by removing validation from the CPU/Memory fields.

@samuelvl can you double check that creating a Notebook with the Jupyter web app also works as expected after this change? To make sure the backend is submitting objects that are valid.

Reply:
I do confirm the backend creates the notebook object correctly after removing the patch:

``` shell
$ oc get notebooks
NAME    AGE
scipy   13m
$ oc get pods     
NAME      READY   STATUS    RESTARTS   AGE
scipy-0   1/1     Running   0          10m
```

This is the notebook created by the JWA:

```yaml
apiVersion: kubeflow.org/v1
kind: Notebook
metadata:
  annotations:
    notebooks.kubeflow.org/last-activity: "2022-04-20T12:06:44Z"
    notebooks.kubeflow.org/server-type: jupyter
  labels:
    app: scipy
  name: scipy
  namespace: kubeflow-user
spec:
  template:
    spec:
      containers:
      - image: public.ecr.aws/j1r0q0g6/notebooks/notebook-servers/jupyter-scipy:v1.5.0
        imagePullPolicy: IfNotPresent
        name: scipy
        resources:
          limits:
            cpu: 600m
            memory: 1288490188800m
          requests:
            cpu: 500m
            memory: 1Gi
        volumeMounts:
        - mountPath: /dev/shm
          name: dshm
        - mountPath: /home/jovyan
          name: scipy-volume
      serviceAccountName: default-editor
      volumes:
      - emptyDir:
          medium: Memory
        name: dshm
      - name: scipy-volume
        persistentVolumeClaim:
          claimName: scipy-volume
```

Reply:
Perfect!